sdk/go/pulumi/context.go

```diff
@@ -966,6 +962,30 @@ func (ctx *Context) collapseAliases(aliases []Alias, t, name string, parent Reso
 		aliasURNs = append(aliasURNs, urn)
 	}
 
+	if parent != nil {
+		parentAliases := parent.getAliases()
+		for i := range parentAliases {
```

Comment:
Nit:

```go
    for i, parentAlias := range parent.getAliases() {
    }
```

Reply:
Unfortunately - this is actually required to be this way because `parentAlias` is captured in a closure, and If written the way you suggest, Go overwrites the captured loop variable (so that all applys see the final element). Several other ways we could write this correctly, but this felt the simplest.

Reply:
Ah yes, by-reference loop variable closure capture, the new 1bln mistake.. Thanks!